metrics/record.go

```diff
@@ -25,14 +25,24 @@ import (
 // TODO should be properly refactored and pieces should move to eventing and serving, as appropriate.
 // 	See https://github.com/knative/pkg/issues/608
 
+// Takes a function that would record a metric, and returns one that only does so if
+// the metric config has been configured.
+func ShouldEmitMetrics() bool {
```

Comment:
is this required to be public?

Reply:
I could move ConfigAware*Metric here, but then they'd need to be public. Or we'd need to add a public method "ConfigureKubemetrics" or something.

Reply:
But where is it used outside the package? Or it's for future use?

Reply:
It's used in controlller/stats_reporter.go, which I may have added after your first round of review.